#!/usr/bin/env python
# _*_ coding:utf-8 _*_

# 记得文件属性要允许作为程序执行文件 #
 
PACKAGE = "robot_sim"      # 这个就是你的include文件的文件夹名字
 
from dynamic_reconfigure.parameter_generator_catkin import *
 
gen = ParameterGenerator()

# 动态参数名字 数据类型 掩码一般为0 说明与描述 参数设置的默认值 可设置的最小值 可设置的最大值 #

gen.add("add_pi",      bool_t,   0, "add_pi", False )
gen.add("good_grasp",  bool_t,   0, "whether good_grasp or not", False )
gen.add("bad_grasp",   bool_t,   0, "whether bad_grasp or not", False )

# gen.add("cmd_pub_rate", int_t,    0, "turtlesim cmd pub rate", 1, 0, 5)
 
# log_enum = gen.enum([ gen.const("info", int_t, 0, "log print flag:INFO"),
#                       gen.const("warn", int_t, 1, "log print flag:WARN"),
#                       gen.const("error", int_t, 2, "log print flag:ERROR")],
#                       "Set Log Level")
# gen.add("log_level", int_t, 0, "Set Log Level", 0, 0, 2, edit_method=log_enum)

# 第二个参数为运行的节点名 第三个参数是生成的文件名字与.cfg文件名字一样 #
exit(gen.generate(PACKAGE, "GPD_method_grasp", "gpd"))    # 用于生成dynamicConfig.h在cpp中可以使用